lightningd/log.c

```diff
@@ -53,10 +53,16 @@ static void log_default_print(const char *prefix,
 			      bool continued,
 			      const char *str, void *arg)
 {
+	struct timeval t;
+	gettimeofday(&t, NULL);
+	char iso8601_s[strlen("YYYY-mm-ddTHH:MM:SS.nnnZ") + 1];
+	size_t offset = strftime(iso8601_s, sizeof(iso8601_s), "%FT%T.", gmtime(&t.tv_sec));
```

Comment:
You could probably have the following `strftime` format: `%FT%T.%%03dZ`, pass it through `strftime` and get a valid `snprintf` string to add the milliseconds. Avoids having to deal with offsets :wink: